### PR TITLE
merge everything into charmcraft.yaml

### PR DESCRIPTION
**Pull request by Thanhphan1147**
Applicable spec: <link>

### Overview

Merge `actions.yaml` , `config.yaml` and `metadata.yaml` into `charmcraft.yaml`

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: charmcraft.yaml [hunk 1]
```diff
@@ -23,3 +23,42 @@ parts:
...
+
+# config
+config:
+  options:
+    external-hostname:
+      default: ""
+      description: The hostname of the gateway.
+      type: string
+
+# actions
```
Comment (arturo-seijas) [inline charmcraft.yaml:57]: Same here

File: charmcraft.yaml [hunk 2]
```diff
@@ -23,3 +23,42 @@ parts:
...
+    - https://launchpad.net/~canonical-is-devops
+assumes:
+  - k8s-api
+  - juju >= 3.3
+requires:
+  certificates:
+    interface: tls-certificates
+    limit: 1
+
+# config
```
Comment (arturo-seijas) [inline charmcraft.yaml:49]: Same here

File: charmcraft.yaml [hunk 3]
```diff
@@ -23,3 +23,42 @@ parts:
       - libssl-dev
       - rustc
       - cargo
+
+# Metadata
```
[the remark this answers is not in the document]
Comment (Thanhphan1147) [reply charmcraft.yaml]: thanks, removed!

File: charmcraft.yaml (hunk 2 above)
Comment (Thanhphan1147) [reply charmcraft.yaml]: thanks, removed!

File: charmcraft.yaml (hunk 1 above)
Comment (Thanhphan1147) [reply charmcraft.yaml]: thanks, removed!

**Review by arturo-seijas — APPROVED** [review]

File: charmcraft.yaml [hunk 4]
```diff
@@ -23,3 +23,39 @@ parts:
...
+  - juju >= 3.3
+requires:
+  certificates:
+    interface: tls-certificates
+    limit: 1
+
+config:
+  options:
+    external-hostname:
+      default: ""
```
Comment (javierdelapuente) [inline charmcraft.yaml:51]: is a default necessary?

**Review by javierdelapuente — APPROVED** [review]

**Review by yanksyoon — APPROVED** [review]

File: charmcraft.yaml (hunk 4 above)
Comment (Thanhphan1147) [reply charmcraft.yaml:51]: good point, I'm not sure if we need a default value here, I'll think about it and make a follow-up PR if needed

**Pull request merged**